### PR TITLE
New intro experience

### PR DESCRIPTION
**Pull request by bemasc**

### CODE REVIEW

File: Android/app/src/main/java/app/intra/IntroDialog.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import androidx.viewpager.widget.ViewPager;
+
+// Fragment representing the full-screen first-run experience.  This intro is launched by the main
+// activity.
+public class IntroDialog extends DialogFragment {
+  private static final int NUM_PAGES = 3;
+
+  // Avoid duplicate instances of this dialog.  This can occur when rotating the app, which would
+  // otherwise cause the main activity to launch an additional instance of this dialog.
+  private static boolean shown = false;
```
Comment (alalamav) [inline Android/app/src/main/java/app/intra/IntroDialog.java]: nit: s/shown/isShown

File: Android/app/src/main/java/app/intra/IntroDialog.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      }
+    });
+
+    pager.addOnPageChangeListener(new TabListener(backButton, nextButton, acceptButton));
+
+    shown = true;
+    return welcomeView;
+  }
+
+  public static class Page0 extends Fragment {
```
Comment (alalamav) [inline Android/app/src/main/java/app/intra/IntroDialog.java]: These class names and duplication seem like a smell to me.
Could we have a single class for the three pages? We could expose a property to set the one different string and in the case of `Page2`, a method to set the `details`?

File: Android/app/src/main/java/app/intra/IntroDialog.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    @Override
+    public int getCount() {
+      return pages.length;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+      return pages[position];
```
Comment (alalamav) [inline Android/app/src/main/java/app/intra/IntroDialog.java]: Maybe check that the index is in bounds?

File: Android/app/src/main/java/app/intra/IntroDialog.java (hunk 1 above)
Comment (bemasc) [reply Android/app/src/main/java/app/intra/IntroDialog.java]: Done

File: Android/app/src/main/java/app/intra/IntroDialog.java (hunk 2 above)
Comment (bemasc) [reply Android/app/src/main/java/app/intra/IntroDialog.java]: OK, I've unified the pages into a single layout and fragment class.  PTAL.

File: Android/app/src/main/java/app/intra/IntroDialog.java (hunk 3 above)
Comment (bemasc) [reply Android/app/src/main/java/app/intra/IntroDialog.java]: Done.

**Comment by alalamav** [comment]
Nice, LGTM!

**Pull request merged**